src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1890,17 +1891,19 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
         INVARIANT(
           !parameters.empty(),
           "GCC polymorphic built-ins should have at least one parameter");
+        /// XXX why are we only looking at the type of the first parameter?
         if(parameters.front().type().id() == ID_pointer)
         {
           identifier_with_type =
             id2string(identifier) + "_" +
-            type2name(parameters.front().type().subtype(), *this);
+            type2identifier(parameters.front().type().subtype(), *this);
         }
         else
         {
           identifier_with_type = id2string(identifier) + "_" +
-                                 type2name(parameters.front().type(), *this);
+                                 type2identifier(parameters.front().type(), *this);
         }
+        identifier_with_type = id2string(identifier_with_type);
```

Comment:
This looks wrong - `identifier_with_type` is an `irep_idt`, so I can understand doing id2string... but then you assign that back to itself...

Reply:
I looked into this. Indeed it looks extraneous. I've removed it.